src/frontends/paddle/src/op/set_value.cpp

```diff
@@ -0,0 +1,120 @@
+// // Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+#define MAX_VALUE(T) std::numeric_limits<T>::max()
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+
+std::shared_ptr<default_opset::Constant> get_max_value_by_dtype(ov::element::Type dtype) {
+    if (dtype == element::f32)
+        return default_opset::Constant::create(dtype, {}, {MAX_VALUE(float)});
+    else
+        return default_opset::Constant::create(dtype, {}, {MAX_VALUE(int)});
+};
+
+Output<Node> handle_minus_index(const OutputVector& node, const Output<Node>& dim) {
+    const auto zero = default_opset::Constant::create(node[0].get_element_type(), {1}, {0});
+    const auto concat_node = std::make_shared<default_opset::Concat>(node, 0);
+    const auto mask = std::make_shared<default_opset::Less>(concat_node, zero);
+    const auto res = std::make_shared<default_opset::Add>(concat_node, dim);
+    return std::make_shared<default_opset::Select>(mask, res, concat_node);
+}
+
+NamedOutputs set_value(const NodeContext& node) {
+    const auto input_node = node.get_input("Input");
+    auto value_node = node.get_input("ValueTensor");
+    const auto input_shape = input_node.get_partial_shape().get_shape();
+    const auto dims = static_cast<int64_t>(input_node.get_partial_shape().rank().get_length());
+    const auto dtype = input_node.get_element_type();
+    const auto axes = node.get_attribute<std::vector<int64_t>>("axes");
+
+    Output<Node> padding_starts_node, padding_ends_node, value_target_shape;
+    Shape value_shape(input_shape);
+
+    if (node.has_input("StartsTensorList") && node.has_input("StepsTensorList") && node.has_input("EndsTensorList")) {
+        const auto dim_node = default_opset::Constant::create(element::i64, {input_shape.size()}, input_shape);
+        auto starts = handle_minus_index(node.get_ng_inputs("StartsTensorList"), dim_node);
+        auto ends = handle_minus_index(node.get_ng_inputs("EndsTensorList"), dim_node);
+        const auto steps = node.get_ng_inputs("StepsTensorList");
+        std::vector<int64_t> needed_input_dim;
+        for (size_t i = 0; i < input_shape.size(); i++) {
+            if (axes[i] == static_cast<int64_t>(i))
```

Comment:
That may be no problem if used in '_setitem_impl_'. What if 'set_value' is not used with '_setitem_impl_' and axes is like this [0, 2]?

Reply:
Sorry, I don't understand what you mean

Reply:
I mean all the testcase use the operator[] to allocate the 'set_value' in paddle. I see that the axes should be like a list [0, 1, 2, 3] in https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/variable_index.py#L641 . What if it is not used with operator[] and set the axes value like [0, 2]?

Reply:
I think we must use operator[] to call `_setitem_impl`. 
[0, 2] is illegal, because we need to specify start and end index in every axis.

Reply:
In fact, `axes` means that there are indexes on these axes. 
For example
```python
a[1:2, 2:3, 3:4]
```
The axes are [0, 1, 2]
```python
a[1:2, :, 3:4]
```
The axes are [0, 2]